### PR TITLE
API: added account related endpoints

### LINKED ISSUE #7967: New Mastodon API endpoints
Here is a list of API endpoints that should be supported next. They do appear in my logfile and they are generated by external statistics tools or clients like Fedilab.

* [x] /api/v1/custom_emojis (#8191)
* [x] /api/v1/notifications?since_id=123456&exclude_types[]=follow&exclude_types[]=favourite&exclude_types[]=reblog&exclude_types[]=poll&limit=30 (#10223)
* [x] /api/v1/timelines/home?limit=20 (#10220)
* [x] /api/v1/timelines/public?local=true&limit=40 (#10220)
* [x] /api/v1/directory?order=active&local=true&limit=20&offset=0 (#9115)
* [x] /api/v1/directory?order=new&local=true&limit=20&offset=0 (#9115)

### PR DESCRIPTION
**Pull request by annando**
Several account related endpoints like the search for accounts or the groups (lists) have been added.

### CODE REVIEW

**Comment by MrPetovan** [comment]
I see the lint check is working overtime! Thanks to @nupplaphil again!

File: src/Factory/Api/Mastodon/ListEntity.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+namespace Friendica\Factory\Api\Mastodon;
+
+use Friendica\BaseFactory;
+use Friendica\Database\DBA;
+
+class ListEntity extends BaseFactory
+{
+	public function create(int $id)
```
Comment (MrPetovan) [inline src/Factory/Api/Mastodon/ListEntity.php]: Please explicit what the list is created from.
```suggestion
	public function createFromGroupId(int $id)
```

File: src/Factory/Api/Mastodon/Notification.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+use Friendica\BaseFactory;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\Contact;
+use Friendica\Model\Notification as ModelNotification;
+
+class Notification extends BaseFactory
+{
+	public function create(int $id)
```
Comment (MrPetovan) [inline src/Factory/Api/Mastodon/Notification.php]: Please explicit what the notification is created from.
```suggestion
	public function createFromNotifyId(int $id)
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Factory/Api/Mastodon/Notification.php (hunk 2 above)
Comment (annando) [reply src/Factory/Api/Mastodon/Notification.php]: Done

File: src/Factory/Api/Mastodon/ListEntity.php (hunk 1 above)
Comment (annando) [reply src/Factory/Api/Mastodon/ListEntity.php]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**